src/Integration.Vsix/SonarLintDaemon/CFamily.cs

```diff
@@ -23,34 +23,81 @@
 using System.IO;
 using System.Reflection;
 using System.Text;
+using EnvDTE;
+using Microsoft.VisualStudio;
 using Newtonsoft.Json;
 
 namespace SonarLint.VisualStudio.Integration.Vsix
 {
-    static class CFamily
+    internal static class CFamily
     {
+        public const string CPP_LANGUAGE_KEY = "cpp";
+        public const string C_LANGUAGE_KEY = "c";
 
-        internal const string CPP_LANGUAGE_KEY = "cpp";
-        internal const string C_LANGUAGE_KEY = "c";
-
-        public static string TryGetConfig(ILogger logger, EnvDTE.ProjectItem projectItem, string absoluteFilePath,
+        public static string TryGetConfig(ILogger logger, ProjectItem projectItem, string absoluteFilePath,
             out string sqLanguage)
         {
+            if (IsHeaderFile(absoluteFilePath))
+            {
+                // We can't analyze header files currently because we can't get all
+                // of the required configuration information
+                logger.WriteLine($"Cannot analyze header files. File: '{absoluteFilePath}'");
```

Comment:
I've added more useful logging messages to explain to the user why the files aren't being analysed (here, and line 51 below).